rules/utils/is-literal-value.js

```diff
@@ -0,0 +1,2 @@
+'use strict';
+module.exports = value => node => node && node.type === 'Literal' && node.value === value;
```

Comment:
I don't think the utility function should be automatically curried. It should just take both value and node as parameters since you will always need both to do the valid comparison.

Personally, the way I'd handle this is to make a base utility for checking node type + value:

```
module.exports = (node, { type, value }) => {
    if (!node) {
        return false;
    }

    if (node.type !== type) {
        return false;
    }

    return node.value !== value;
}
```

Then use that to create a specific literal check:

```
module.exports = (node, { value }) => {
    return is-node-match(node, { type: 'Literal', value });
}
```

But really, the farther you go down this path the more you just want a deep object comparison utility.

Reply:
this code is  from  https://github.com/sindresorhus/eslint-plugin-unicorn/blob/a2469ade9d2349fc74b9ed0ce67024cb3404b6b8/rules/no-for-loop.js#L5

I was  trying to  write

```js
const isLiteralValue = (node, value) => node && node.type === 'Literal' && node.value === value;

const isLiteralZero = node  => isLiteralValue(node, 0)
```

but curry seems easier to use, wish we can  use [Partial Application]https://github.com/tc39/proposal-partial-application) much easier to use 

```js
const isLiteralZero = isLiteralValue(?, 0)
```